src/sass/parse.js

```diff
@@ -1115,15 +1115,30 @@ function getBrackets() {
 function checkClass(i) {
   let start = i;
   let l;
+  let wasIdent = true;
 
   if (i >= tokensLength) return 0;
 
   if (tokens[i].class_l) return tokens[i].class_l;
 
   if (tokens[i++].type !== TokenType.FullStop) return 0;
 
-  if (l = checkIdentOrInterpolation(i)) i += l;
-  else return 0;
+  // If a class is prefixed with `-` only allow if it contains an ident later
+  if (tokens[i].type === TokenType.HyphenMinus) wasIdent = false;
+
+  while (i < tokensLength) {
+    if (l = checkIdentOrInterpolation(i)) {
+      wasIdent = true;
+      i += l;
+    }
+    else if (tokens[i].type === TokenType.HyphenMinus) i += 1;
+    else break;
+  }
```

Comment:
Looks good to me. Have you found a detailed spec for valid css selectors? I can't find anything in the docs!